### PR TITLE
Add `amazons3://` backend.

### PR DESCRIPTION
**Pull request by waaaaargh**
This PR adds a new `amazons3://` backend.

For mor information regarding the S3 please see the the [README.md](https://github.com/waaaaargh/knoxite/tree/backend/amazons3/storage/amazons3)

TODO before merging:
- [x] Deploy [test infrastructure][1] in knoxite AWS Account
- [x] Update [required secrets][2] on knoxite/knoxite

[1]: https://github.com/waaaaargh/knoxite/blob/backend/amazons3/storage/amazons3/test_setup.template.yaml
[2]: https://github.com/waaaaargh/knoxite/blob/backend/amazons3/.github/workflows/test.yml

### CODE REVIEW

File: storage/amazons3/amazons3.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
+/*
+ * knoxite
+ *     Copyright (c) 2016, Johannes Fürmann <fuermannj+floss@gmail.com>
```
Comment (craftamap) [inline storage/amazons3/amazons3.go:3]: You may want to update the year, here :D

Comment (waaaaargh) [reply storage/amazons3/amazons3.go]: Thanks for the hint, fixed.

File: storage/amazons3/README.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+| `region` | valid AWS region descriptors | AWS Region. If this configuration is not specified, the backend falls back to other means of configuration, such as the `AWS_REGION` environment variable. |
+| `endpoint` | valid URLs | **For testing purposes only**. This Parameter can be used to make S3 requests against backends other than those provided by AWS. This is not recommended.
+| `force_path_style` | `true` | Use this parameter to force the underlying S3 SDK to make "path style" requests against the Amazon S3 backend. We don't recommend using this parameter if not required for compatibility reasons as [path style request are being sunset by AWS.][1]
+
+
+## S3 bucket setup
+
+For security reasons, we recommend setting up an S3 bucket and using short-lived credentials such as EC2 instance roles whenever feasible.
+
+In situations where this is not possible, long-lived credentials such as IAM users are a good options.
```
Comment (ma-hartma) [inline storage/amazons3/README.md:22]:
```suggestion
In situations where this is not possible, long-lived credentials such as IAM users are a good option.
```

**Review by ma-hartma — APPROVED** [review]

**Review by muesli — APPROVED** [review]
Yay, LGTM!

**Review by penguwin — APPROVED** [review]

**Review by craftamap — APPROVED** [review]

**Comment by craftamap** [comment]
Nice, lets get this merged!

**Pull request merged**